smart-cards/src/main/java/gov/va/api/health/smartcards/patient/PatientController.java

```diff
@@ -83,13 +80,24 @@ void consumeBundle(B bundle, List<MixedEntry> target, Function<E, MixedEntry> mi
     bundle.entry().stream().map(minimizer).forEachOrdered(target::add);
   }
 
-  private static List<CredentialType> credentialTypes(Parameters parameters) {
+  static List<CredentialType> credentialTypes(Parameters parameters) {
     checkRequestState(parameters.parameter() != null, "parameters are required");
-    return parameters.parameter().stream()
-        .filter(p -> "credentialType".equals(p.name()))
-        .map(Parameters.Parameter::valueUri)
-        .map(CredentialType::fromUri)
-        .collect(toList());
+    List<CredentialType> types =
+        parameters.parameter().stream()
+            .filter(p -> "credentialType".equals(p.name()))
+            .map(Parameters.Parameter::valueUri)
+            .map(CredentialType::fromUri)
+            .collect(toList());
+    validateCredentialTypes(types);
+    // Expand credential types.
+    if (!types.contains(CredentialType.HEALTH_CARD)) {
```

Comment:
Would it make sense to make `types` a set instead of a list? That would simplify this check and the one below.

(`collect(toSet())` for types would also take care of eliminating duplicate credentialTypes)

Reply:
Yes, good call on using Set instead.